back-end/app/controllers/book_controller.js

```diff
@@ -51,7 +51,16 @@ exports.getMatchingBooks= async (info) => {
 	}
 	query += `;`;
     result = await db.query(query, params);
-	return result[0];
+    let data = result[0];
+    for (let index = 0; index < data.length; index++) {
+        let bookRecord = data[index];
+        if (bookRecord.authors) {
+            bookRecord.authors = bookRecord.authors.split(",");
+        } else {
+            bookRecord.authors = [];
```

Comment:
Do you also need to run a genre check here like you did on LIne 97? I don't have experience with this so I'm just checking.

Reply:
We don't only because we don't actually return any genre information to the front-end in the search endpoint which this method is for. The detail endpoint does return genre information and that is why line 97 does this thing for genres.